tests/e2e/support/objects/app-files/spaces/actions.ts

```diff
@@ -206,3 +208,45 @@ export const openSpaceTrashBin = async (args: openSpaceTrashBinArgs): Promise<vo
   await page.locator(spaceContextButton).click()
   await page.locator(spaceDeletedFilesButton).click()
 }
+
+export interface removeAccessMembersArgs {
+  users: User[]
+  page: Page
+}
+export const removeAccessSpaceMembers = async (args: removeAccessMembersArgs): Promise<void> => {
+  const { page, users } = args
+  await sidebar.open({ page: page })
+  await sidebar.openPanel({ page: page, name: 'space-share' })
+  await removeSharee({ page, users })
+}
+
+export interface searchForSpacesIdsArgs {
+  spaceID: string
+  page: Page
+}
+export const searchForSpacesIds = async (args: searchForSpacesIdsArgs): Promise<boolean> => {
+  const { page, spaceID } = args
+  const emptySpacesMessage = "You don't have access to any spaces"
+
+  // for empty case
+  if (await page.$(util.format(emptySpacesSelector, emptySpacesMessage))) {
```

Comment:
I remember that we refused to use `page.$` as it causes flaky. 
https://playwright.dev/docs/api/class-page#page-query-selector

Can we do it differently instead?
Sorry, i didn't understand how it works. I didn't found locator emptySpacesSelector

Reply:
I will look in to it to do differently.
@ScharfViktor it works as, when users is added to single space and is removed then the empty message is displayed when the user navigates to his/her respective space, and in case of multiple space, it checks only the space id, if does not exists returns `true` for both case.